src/main/resources/avro/readmethods.avdl

```diff
@@ -14,8 +14,10 @@ to `ReferenceSet`s containing that same `Reference`. If no reference is
 specified, all `ReadGroup`s must be aligned to the same `ReferenceSet`.
 */
 record SearchReadsRequest {
-  /** If nonempty, restrict this query to reads within the given readgroups. */
-  array<string> readGroupIds = [];
+  /**
+  The ReadGroup to search.
+  */
+  string readGroupId;
```

Comment:
@jeromekelleher your understanding is correct. I agree with @lh3 that using the proposed workaround will in many cases require pushing complexity down to the client. I also think that the statement about negligible overhead is predicated on assumptions about the implementation, e.g. how the data is stored and ordered on disk. Probably for a BAM-backed implementation you would expect similar performance with either approach.